### PR TITLE
feat(dataset): add class "Commit" to support basic version control

### PR DESCRIPTION
**Pull request by graczhual**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2k67ayq 版本管理基础类：Commit](https://app.clickup.com/t/2k67ayq)

File: graviti/dataset/struct.py [hunk 1]
```diff
@@ -55,3 +56,87 @@ def edit(self, title: Optional[str] = None, description: Optional[str] = None) -
...
+        committed_at: The time when the draft is committed.
+
+    """
+
+    _T = TypeVar("_T", bound="Commit")
+
+    _repr_attrs: Tuple[str, ...] = ("parent_commit_id", "title", "committer", "committed_at")
+
+    commit_id: str = attr(key=camel)
+    parent_commit_id: Optional[str] = attr(key=camel)
```
Comment (linjiX) [inline graviti/dataset/struct.py:79]: do we still need `key=camel`?

File: graviti/dataset/struct.py [hunk 2]
```diff
@@ -55,3 +56,87 @@ def edit(self, title: Optional[str] = None, description: Optional[str] = None) -
...
+        self.title = title
+        self.description = description
+        self.committer = committer
+        self.committed_at = committed_at
+
+    def _repr_head(self) -> str:
+        return f'{self.__class__.__name__}("{self.commit_id}")'
+
+    @classmethod
+    def loads(cls: Type[_T], contents: Dict[str, Any]) -> _T:
```
Comment (linjiX) [inline graviti/dataset/struct.py:105]: we do not use `loads` and `dumps` in graviti SDK
use `from_xxx`, `to_xxx` instead

File: graviti/dataset/struct.py [hunk 3]
```diff
@@ -55,3 +56,87 @@ def edit(self, title: Optional[str] = None, description: Optional[str] = None) -
...
+
+    def dumps(self) -> Dict[str, Any]:
+        """Dumps all the commit information into a dict.
+
+        Returns:
+            A dict containing all the information of the commit::
+
+                {
+                    "commitId": <str>
+                    "parentCommitId": <str> or None
```
Comment (linjiX) [inline graviti/dataset/struct.py:134]: `parent_commit_id`?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataset/struct.py [hunk 4]
```diff
@@ -55,3 +56,87 @@ def edit(self, title: Optional[str] = None, description: Optional[str] = None) -
...
+        committed_at: The time when the draft is committed.
+
+    """
+
+    _T = TypeVar("_T", bound="Commit")
+
+    _repr_attrs: Tuple[str, ...] = ("parent_commit_id", "title", "committer", "committed_at")
+
+    commit_id: str = attr()
+    parent_commit_id: str = attr(default="")
```
Comment (linjiX) [inline graviti/dataset/struct.py:79]: should the `parent_commit_id` default to `""`

Comment (linjiX) [reply graviti/dataset/struct.py:79]: @Lee-000 please help on this

**Review by linjiX — APPROVED** [review]

**Pull request merged**